solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -132,8 +132,9 @@ protected Http2SolrClient(String serverBaseUrl, Builder builder) {
       if (builder.followRedirects != null
           || builder.connectionTimeoutMillis != null
           || builder.maxConnectionsPerHost != null
-          || builder.useHttp1_1
-              != builder.httpClient.getTransport() instanceof HttpClientTransportOverHTTP
+          || (builder.providedUseHttp1_1 != null
+              && builder.providedUseHttp1_1
+                  != builder.httpClient.getTransport() instanceof HttpClientTransportOverHTTP)
```

Comment:
I think we should not drop this part now -- just keep this looking _just_ like the other settings with null checks.